content/en/blog/_posts/2024-04-17-kubernetes-1.30.md

```diff
@@ -0,0 +1,162 @@
+---
+layout: blog
+title: 'Kubernetes v1.30: TBA'
+date: 2024-04-17
+slug: kubernetes-v1-30-release
+---
+
+**Authors:** [Kubernetes v1.30 Release Team](https://github.com/kubernetes/sig-release/blob/master/releases/release-1.30/release_team.md)
+
+**Editors:** Amit Dsouza, Frederick Kautz, Kristin Martin, Abigail McCarthy, Natali Vlatko
+
+Announcing the release of Kubernetes v1.30: Theme TBA, the first release of 2024!
+
+Similar to previous releases, the release of Kubernetes v1.30 introduces new stable, beta, and alpha
+features. The consistent delivery of top-notch releases underscores the strength of our development
+cycle and the vibrant support from our community.
+
+This release consists of XX enhancements. Of those enhancements, XX have graduated to Stable, XX are
+entering Beta, and XX have graduated to Alpha.
+
+
+## Release theme and logo
+<Logo image size is recommended to be no more than 2160px> 
+[Kat requested to hold review for this section until after the theme is revealed, and other 
+sections are reviewed.]
+
+
+## Improvements that graduated to stable in Kubernetes v1.30 {#graduations-to-stable}
+
+_This is a selection of some of the improvements that are now stable following the v1.30 release._
+
+### Robust VolumeManager reconstruction after kubelet restart ([SIG Storage](https://github.com/kubernetes/community/tree/master/sig-storage))
+
+Blurb for 3756 goes here.
+
+### Prevent unauthorised volume mode conversion during volume restore ([SIG Storage](https://github.com/kubernetes/community/tree/master/sig-storage))
+
+Blurb for 3141 goes here.
+
+### Pod Scheduling Readiness ([SIG Scheduling](https://github.com/kubernetes/community/tree/master/sig-scheduling))
+
+Blurb for 3521 goes here.
+
+### Min domains in PodTopologySpread ([SIG Scheduling](https://github.com/kubernetes/community/tree/master/sig-scheduling))
+
+Blurb for 3022 goes here.
+
+### Go workspaces for k/k ([SIG Architecture](https://github.com/kubernetes/community/tree/master/sig-architecture))
+
+Blurb for 4402 goes here.
+
+## Improvements that graduated to beta in Kubernetes v1.30 {#graduations-to-beta}
+
+_This is a selection of some of the improvements that are now beta following the v1.30 release._
+
+### Node log query ([SIG Windows](https://github.com/kubernetes/community/tree/master/sig-windows))
+
+Blurb for 2258 goes here.
+
+### CRD Validation Ratcheting ([SIG API Machinery](https://github.com/kubernetes/community/tree/master/sig-api-machinery))
+
+Blurb for 4008 goes here--subject to Docs Freeze
+
+### Contextual logging ([SIG Instrumentation](https://github.com/kubernetes/community/tree/master/sig-instrumentation))
+
+Blurb for 3077 goes here.
+
+## New alpha features
+
+### Speed up recursive SELinux label change ([SIG Storage](https://github.com/kubernetes/community/tree/master/sig-storage))
+
+Blurb for 1710 goes here.
```

Comment:
@jsafrane Looks like this one also needs a short paragraph:).

Reply:
This is a niche and complex feature. My attempt:

```suggestion
This feature allows kubelet to apply SELinux label to volumes used in a pod in constant time using a mount option,  which speeds up pod startup on nodes with SELinux enabled. Without this feature, the container runtime had to recursively walk through the whole volumes and apply SELinux label individually to each file, which can be slow for volumes with large amount of files and directories.

Kubernetes 1.27 graduated this feature as beta, but limited it to ReadWriteOncePod volumes. The corresponding feature gate is `SELinuxMountReadWriteOncePod`. It remains beta in 1.30.

Kubernetes 1.30 extends support for SELinux mount option to all volumes as alpha, with a separate feature gate `SELinuxMount`. This feature gate introduces a behavioral change when multiple Pods with different SELinux labels share the same volume. See [KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1710-selinux-relabeling#behavioral-changes) for details. We strongly encourage users that run  Kubernetes with SELinux enabled to test this feature and provide any feedback on the KEP.

| Feature gate | Stage in v1.30 | Behavior change |
|---|---|---| 
| SELinuxMountReadWriteOncePod |  Beta | No |
| SELinuxMount | Alpha | Yes | 

Both feature gates `SELinuxMountReadWriteOncePod` and `SELinuxMount` must be enabled to test this feature on all volumes.

This feature has no effect on Windows nodes or on Linux nodes without SELinux support.